src/Artifacts.jl

```diff
@@ -673,8 +673,10 @@ function bind_artifact!(artifacts_toml::String, name::String, hash::SHA1;
     end
 
     # Spit it out onto disk
-    open(artifacts_toml, "w") do io
-        TOML.print(io, artifact_dict, sorted=true)
+    let artifact_dict = artifact_dict
```

Comment:
I'm surprised this is needed, there is no assignment to `artifact_dict` so shouldn't the frontend be able to figure this out?

Reply:
I'm not a hundred percent certain but the codegen shows some try/catch stuff that appears to complicate the analysis. In practice, adding this suddenly gave much nicer codegen.